### ISSUE TITLE
PushNotificationIOS Documentation Has Step that Leads to Error

### ISSUE THREAD

**Issue by mikeborozdin** [facebook/react-native#5162]
The [documentation of PushNotificationIOS]( https://facebook.github.io/react-native/docs/pushnotificationios.html#content) says one needs to add

```
   // Required to register for notifications
   - (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings
   {
    [RCTPushNotificationManager didRegisterUserNotificationSettings:notificationSettings];
   }
   // Required for the register event.
   - (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
   {
    [RCTPushNotificationManager didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
   }
   // Required for the notification event.
   - (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)notification
   {
    [RCTPushNotificationManager didReceiveRemoteNotification:notification];
   }
```

to `AppDelegate.m`

However, the first method, namely

```
   - (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings
   {
    [RCTPushNotificationManager didRegisterUserNotificationSettings:notificationSettings];
   }
```

results in an compilation error that says

>No class method for selector 'didRegisterUserNotificationSettings:notificationSettings'

**Comment by trave7er** [comment]
You are probably using an older version of ReactNative, if you go to `RCTPushNotificationManager.h` file, you will be able to see the actual available methods for your version.

**Comment by hernanmateo** [comment]
Same error here

**Comment by kushal** [comment]
Afaict, these methods all require application for the first argument, and the documentation, is just wrong? i.e. 
```
// Required to register for notifications
- (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings
{
  [RCTPushNotificationManager application:application didRegisterUserNotificationSettings:notificationSettings];
}
// Required for the register event.
- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
{
  [RCTPushNotificationManager application:application didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
}
// Required for the notification event.
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)notification
{
  [RCTPushNotificationManager application:application didReceiveRemoteNotification:notification];
}
```
wfm

**Comment by daemonchen** [comment]
same error

**Comment by rollick** [comment]
For the latest versions of react-native, see here:

https://github.com/facebook/react-native/blob/1dc5dca604c9973179e137f226418ca89e28b58a/Libraries/PushNotificationIOS/PushNotificationIOS.js#L42-L56

**Comment by tehong** [comment]
same error here.

**Comment by yamill** [comment]
I'm having the same error

**Comment by jihopark** [comment]
https://github.com/facebook/react-native/blob/1dc5dca604c9973179e137f226418ca89e28b58a/Libraries/PushNotificationIOS/PushNotificationIOS.js#L42-L56 This solves. the problem @rollick 

but why isn't this mentioned in the Release note? it is a breaking change, no?

**Comment by yamill** [comment]
I have a fix for this. I'll be a submitting a PR sometime soon.

**Comment by scgough** [comment]
Can I confirm what solves the problem @rollick @jihopark ?

I have the following in my `AppDelegate.m` file but still have the `No known class method for selector ###` error on each of those methods.

```
#import "RCTPushNotificationManager.h"
...
// Required to register for notifications
- (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings
{
  [RCTPushNotificationManager didRegisterUserNotificationSettings:notificationSettings];
}
// Required for the register event.
- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
{
  [RCTPushNotificationManager didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
}

// Required for the notification event.
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)notification
{
  [RCTPushNotificationManager didReceiveRemoteNotification:notification];
}
```

**Comment by christopherdro** [comment]
@joshuagutierrez What's the resolution here?

**Comment by joshuagutierrez** [comment]
Sorry, this issue is still open? I got confused when you closed #5746, I read it as "5746 is closed, and it also fixes the same problem as 5162" - I see you mean that "5746 re-asks the open issue 5162"

**Comment by grabbou** [comment]
I think it was just an issue of docs being out of sync with the current master (when we didn't have per-version docs).

As of current master and 0.24 - [the documentation](https://github.com/facebook/react-native/blob/master/Libraries/PushNotificationIOS/PushNotificationIOS.js#L48-L67) does not have `application:(UIApplication *)application` which matches the [native methods](https://github.com/facebook/react-native/blob/master/Libraries/PushNotificationIOS/RCTPushNotificationManager.h#L16-L19)

That was the commit that changed the payloads https://github.com/facebook/react-native/commit/1dc5dca604c9973179e137f226418ca89e28b58a
